doc/safegcd_implementation.md

```diff
@@ -126,6 +126,33 @@ The approach here avoids the need for 2x2 matrix multiplications of various size
 be faster at the level of optimization we're able to do in C.
 
 
+## 2b. From GCDs to Jacobi symbol
```

Comment:
I think this section may be a bit more understandable if it's placed after the other ones it's referencing.

Reply:
Good point! Moved it to the end